tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/observability/tracing/TracingBaseTestCase.java

```diff
@@ -67,9 +67,11 @@ public void cleanup() throws Exception {
         super.cleanupServer();
     }
 
-    protected List<BMockSpan> getFinishedSpans(String serviceName, String resource) throws IOException {
+    protected List<BMockSpan> getFinishedSpans(String serviceName, String entrypointModule,
+                                               String entrypointPosition) throws IOException {
         return getFinishedSpans(serviceName).stream()
-                .filter(span -> Objects.equals(span.getTags().get("resource"), resource))
+                .filter(span -> Objects.equals(span.getTags().get("entrypoint.function.module"), entrypointModule))
+                .filter(span -> Objects.equals(span.getTags().get("entrypoint.function.position"), entrypointPosition))
```

Comment:
This can be a single `filter` call with `&&` operator.

Reply:
FIxed with 3a5fad4b8ee95545ae441daea7a28e24052367f6